TLM/TLM/LoadingExtension.cs

```diff
@@ -333,21 +347,37 @@ public class ManualHarmonyPatch {
                 Flags.OnLevelUnloading();
                 GlobalConfig.OnLevelUnloading();
 
+                var gameObject = UIView.GetAView().gameObject;
+                void Destroy<T>() where T : MonoBehaviour {
+                    Object obj = (Object)gameObject.GetComponent<T>();
+                    if (obj != null) {
+                        Object.Destroy(obj);
+                    }
+                }
+
                 // remove vehicle button
-                var removeVehicleButtonExtender = UIView
-                                                  .GetAView().gameObject
-                                                  .GetComponent<RemoveVehicleButtonExtender>();
-                if (removeVehicleButtonExtender != null) {
-                    Object.Destroy(removeVehicleButtonExtender, 10f);
+                Destroy<RemoveVehicleButtonExtender>();
+                Destroy<RemoveCitizenInstanceButtonExtender>();
+
+                //simManager?.Remove(CustomPathManager);
+
+                //Object.Destroy(CustomPathManager);
+                //CustomPathManager = null;
+
+                Log.Info("Removing Controls from UI.");
+                if (BaseUI != null) {
+                    BaseUI.Close();
```

Comment:
Possibly won't matter as it's only devs who are going to use hot reload, but does `BaseUI.Close()` clear all event handlers from any active UI? If not, `Object.Destroy(BaseUI)` is probably going to mem leak. CO's UI stuff doesn't ditch event handlers on destroy.